File: sdks/swift/Sources/PuxbaySDK/Models/PurchaseOrder.swift
import Foundation

public struct PurchaseOrder: Codable {
    public let id: String
    public let poNumber: String
    public let supplier: String
    public let supplierName: String?
    public let status: String
    public let totalAmount: Double
    public let expectedDeliveryDate: String?
    public let items: [[String: String]]? // Or explicit POItem struct if we knew structure perfectly
    public let createdAt: String
    public let updatedAt: String
}

File: sdks/rust/src/models/purchase_order.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct PurchaseOrder {
    pub id: String,
    pub po_number: String,
    pub supplier: String,
    pub supplier_name: Option<String>,
    pub status: String,
    pub total_amount: f64,
    pub expected_delivery_date: Option<String>,
    // Simplified items for now, ideally strictly typed POItem
    pub items: Option<Vec<serde_json::Value>>, 
    pub created_at: String,
    pub updated_at: String,
}
